packages/flutter_tools/lib/src/commands/screenshot.dart

```diff
@@ -32,39 +52,117 @@ class ScreenshotCommand extends FlutterCommand {
 
   @override
   Future<int> verifyThenRunCommand() async {
-    device = await findTargetDevice();
-    if (device == null)
-      return 1;
+    if (argResults[_kSkia] != null) {
+      if (argResults[_kOut] != null && argResults[_kSkiaServe] != null) {
+        printError('Cannot specify both --$_kOut and --$_kSkiaServe');
+        return 1;
+      }
+    } else {
+      if (argResults[_kSkiaServe] != null) {
+        printError('Must specify --$_kSkia with --$_kSkiaServe');
+        return 1;
+      }
+      device = await findTargetDevice();
+      if (device == null) {
+        printError('Must specify --$_kSkia or have a connected device');
+        return 1;
+      }
+      if (!device.supportsScreenshot) {
```

Comment:
You'll want to make this check specific to when they're taking a screenshot, and not when using --skia.